src/game/assets/index.js

```diff
@@ -5,11 +5,67 @@ import { tilesetRowCount } from '../scenes/LevelScene/tileRules';
+  kingRun02: new Sprite(kingSpritesUrl, new Crop(new Vector(2584, 16), 37, 30)),
+  kingRun03: new Sprite(kingSpritesUrl, new Crop(new Vector(2661, 17), 37, 29)),
+  kingRun04: new Sprite(kingSpritesUrl, new Crop(new Vector(2739, 20), 37, 26)),
+  kingRun05: new Sprite(kingSpritesUrl, new Crop(new Vector(2817, 18), 37, 28)),
+  kingRun06: new Sprite(kingSpritesUrl, new Crop(new Vector(2895, 16), 37, 30)),
+  kingRun07: new Sprite(kingSpritesUrl, new Crop(new Vector(2973, 17), 37, 29)),
+  kingRun08: new Sprite(kingSpritesUrl, new Crop(new Vector(3051, 20), 37, 26)),
+
+  kingAttack01: new Sprite(kingSpritesUrl, new Crop(new Vector(3140, 0), 58, 46)),
+  kingAttack02: new Sprite(kingSpritesUrl, new Crop(new Vector(3219, 5), 51, 41)),
+  kingAttack03: new Sprite(kingSpritesUrl, new Crop(new Vector(3285, 16), 37, 30)),
+
+  kingDoorOut01: new Sprite(kingSpritesUrl, new Crop(new Vector(639, 5), 26, 41)),
+  kingDoorOut02: new Sprite(kingSpritesUrl, new Crop(new Vector(718, 3), 26, 43)),
+  kingDoorOut03: new Sprite(kingSpritesUrl, new Crop(new Vector(797, 5), 26, 41)),
+  kingDoorOut04: new Sprite(kingSpritesUrl, new Crop(new Vector(874, 3), 26, 43)),
+  kingDoorOut05: new Sprite(kingSpritesUrl, new Crop(new Vector(952, 5), 26, 41)),
+  kingDoorOut06: new Sprite(kingSpritesUrl, new Crop(new Vector(1021, 13), 37, 33)),
+  kingDoorOut07: new Sprite(kingSpritesUrl, new Crop(new Vector(1099, 14), 37, 32)),
+  kingDoorOut08: new Sprite(kingSpritesUrl, new Crop(new Vector(1179, 20), 37, 26)),
+
+  kingDoorIn01: new Sprite(kingSpritesUrl, new Crop(new Vector(6, 1), 37, 45)),
+  kingDoorIn02: new Sprite(kingSpritesUrl, new Crop(new Vector(94, 6), 26, 40)),
+  kingDoorIn03: new Sprite(kingSpritesUrl, new Crop(new Vector(175, 5), 29, 41)),
+  kingDoorIn04: new Sprite(kingSpritesUrl, new Crop(new Vector(254, 6), 29, 40)),
+  kingDoorIn05: new Sprite(kingSpritesUrl, new Crop(new Vector(331, 5), 29, 41)),
+  kingDoorIn06: new Sprite(kingSpritesUrl, new Crop(new Vector(409, 6), 29, 40)),
+  kingDoorIn07: new Sprite(kingSpritesUrl, new Crop(new Vector(488, 5), 29, 41)),
+  kingDoorIn08: new Sprite(kingSpritesUrl, new Crop(new Vector(565, 6), 29, 40)),
+
+  kingJump: new Sprite(kingSpritesUrl, new Crop(new Vector(2427, 15), 37, 31)),
+  kingFall: new Sprite(kingSpritesUrl, new Crop(new Vector(1257, 15), 37, 31)),
+  kingGround: new Sprite(kingSpritesUrl, new Crop(new Vector(1335, 20), 37, 26)),
+
+  kingDead01: new Sprite(kingSpritesUrl, new Crop(new Vector(3360, 18), 37, 28)),
+  kingDead02: new Sprite(kingSpritesUrl, new Crop(new Vector(3434, 25), 38, 21)),
+  kingDead03: new Sprite(kingSpritesUrl, new Crop(new Vector(3511, 26), 38, 20)),
+  kingDead04: new Sprite(kingSpritesUrl, new Crop(new Vector(3588, 27), 38, 19)),
+
+  kingHit01: new Sprite(kingSpritesUrl, new Crop(new Vector(1411, 14), 37, 32)),
+  kingHit02: new Sprite(kingSpritesUrl, new Crop(new Vector(1490, 15), 37, 31)),
+
```

Comment:
Wracając trochę do poprzedniego komentarza, teraz jak jest możliwość zmiany pozycji `BoxCollider`a bez zmieniania rozmiaru `Sprite`'ów i `transform`a zastanowiłbym się nad tym. Zaoszczędzisz kilkadziesiąt linijek bardzo powtarzalnego kodu, a przy okazji kilka innych problemów rozwiąże się z automatu, w szczególności, zauważ, że teraz przy ataku:
- król "teleportuje się" w ciągu jednej klatki w lewo, a młot wymachuje w jego oryginalnej pozycji (zamiast braku przemieszczenia i wymachu przed postacią)
- ta "poświata" ataku jest też w `Collider`ze, więc, przykładowo, jak już będziemy mieli świnkę, która zaatakuje młot to stracimy zdrowie (mimo że nas nie trafiła)
- wejdź w obszar wysokości 1 (na prawo od drugiej świnki) i przytrzymaj X
Wszystkie te bugi wynikają z miksu powiązania 1:1 rozmiaru `Collider`a z rozmiarem `Sprite` i zmieniania środka cieżkości (co jest w kontekście fizyki rozważającej obiekt tylko na podstawie środka ciężkości tym samym co teleportacja)

Reply:
Poprawione 👍. Jedyne collider martwego króla się trochę nie zgadza, ale wtedy i tak gra się kończy.

Reply:
Na razie ustawiłam collider obejmujący samego króla (bez młota), co jest okej w przypadku kolizji ze świnką, ale średnio wygląda podczas kolizji z blokiem - można pomyśleć, żeby w przyszłości nałożyć drugi collider na młot 🤔

Reply:
> ale średnio wygląda podczas kolizji z blokiem

Wydaje mi się że większość gier pozwala na przenikanie tego przez ścianę, ale trzeba by zrobić głębszy research żeby to potwierdzić.

Tak czy siak myślę, że będzie ok, bo gameplay jest przyjemniejszy (łatwiej się zmieścić w "szparze", nie będziemy uderzeni niesprawiedliwie przez przeciwnika, nie dzieją się dziwne rzeczy kiedy stoimy przy ścianach), nawet jeśli jest to trochę mniej realistyczne.